### DISCUSSION TITLE
redirect and the 404 page with a link

### DISCUSSION THREAD

**Discussion by amrutadotorg** [sqlpage/SQLPage#415] [Q&A]
Hi, I would like to show a 404 error page to a user and display the link they clicked. After the redirect, it seems variables are not accessible. Here is my current code:

**Code:**

```sql
SELECT 'redirect' AS component, '404.sql' AS link
WHERE NOT EXISTS (SELECT 1 FROM language_check);
```

404.sql
```sql
SELECT 'dynamic' AS component,
       sqlpage.run_sql('shell.sql') AS properties;

SELECT 
    'alert' AS component,
    'Error' AS title,
    'Page not found' AS description,
    'alert-circle' AS icon,
    'red' AS color;
```

**Comment by lovasoa** [comment]
What do you mean by "variables are not accessible" ?

If you loaded `/page.sql?x=1`, and `page.sql` used the redirect component to redirect to `404.sql`, then the user would be redirected to `/404.sql`, not `/404.sql?x=1`. Is this what you meant ?

You can use [sqlpage.path](https://sql.ophir.dev/functions.sql?function=path#function) and [sqlpage.variables](https://sql.ophir.dev/functions.sql?function=variables#function) to build a custom redirect string that will contain the original page's parameters if you want.

**Comment by amrutadotorg** [comment]
ok, thank you I will try
